README.md

```diff
@@ -72,27 +72,148 @@ Pingdom will be integrated with
+- `user-ids.conf` contains Pingdom user IDs.
+   User IDs that are not shared can be placed in the `*.ping` health checks directly.
 
 ### Create a new check or update an existing
-Update the `build.sh` to include a new check definition or change an existing check.
-Updates to checks are performed by _name_. Name changes are seen as new checks.
+Create or update `pingdom-checks/*.ping` file to include a new check definition or change an 
+existing check.
+
+> Pro Tip: Create a curl command of the check _first_. Once you done this, translating it
+> to Pingdom is fairly straight forward.
+
+Pingdom checks use a _template_ to create the HTTP request. Many checks follow the same pattern,
+so it is likely a template already exists to meet your needs. If not, create a new template
+in Pingdom API format.
+
+> Warning: As of March 2020, the Pingdom API incorrectly documents request headers structure.
+> Instead of a _map_ of headers, you must create a field named `requestheader<Name>`, e.g.
+> ```
+> "requestheaderAuthorization":"Authorization:Bearer $authorization_token",
+> "requestheaderAccept":"Accept:application/json+fhir",
+> ```
+
+
+Updates to checks are performed by _name_. Name changes are seen as new checks. If you need
+to change the name, you will need to delete the previous check manually.
+
+#### Template processing
+Templates include variables that are substituted as the checks are processed. For example, 
+```
+{
+  "name": "$name",
+  "type": "http",
+  "host": "$host",
+  ...
+```
+The check definition must include _all_ variables or it will be rejected.
+
+#### Using `bin/pingdom`
+The pingdom utility is automatically included in the path available to your checks.
+You can use it manually also. See `bin/pingdom --help`
+
+##### Saving checks
+The `save-check` command is used to create or update checks in Pingdom. 
+You must specify a template (by name) and template arguments for processing. 
+```
+pingdom save-check \
+  --template awesome-check \
+  -a name=production-awesomeness \
+  -a host=api.va.gov \
+  ...
+```
+You may _test_ templating with the `process-template` command.
+```
+$ bin/pingdom process-template --template https-public-200 -a name=awesome -a url=awesome.com
+Missing template argument: port
+Missing template argument: host
+Missing template argument: resolution
+Missing template argument: group
+Missing template argument: responsetime_threshold
+Missing template argument: userids_csv
+Missing template argument: integrationids_csv
+Missing parameters for template: https-public-200
+``` 
+> Pro Tip: If a template requires an integration or user argument but you do not need one,
+> you may use `$NO_INTEGRATIONS` or `$NO_USERS` to satisfy the template. 
+
+
 
 
 ### Running locally
-if `secrets.conf` file is present, it will be sourced.
+
+> Note: Since there is only one Pingdom instance, testings changes locally is difficult.
+A trial account can be used.
```

Comment:
~~Maybe a blurb about our _hack_ for finding the integration id of a new trial account?~~

Link to the blurb below?

Reply:
/shrug